File: fixtures/scale/scala/large/src/cryptobench/verify/Determinism.scala
package cryptobench.verify

import cryptobench.core.HashFunction

object Determinism {
  def stable(hash: HashFunction, sample: String): Boolean =
    hash.digest(sample) == hash.digest(sample)

  def distinct(hash: HashFunction, left: String, right: String): Boolean = {
    if (left == right) true
    else hash.digest(left) != hash.digest(right)
  }
}

File: fixtures/scale/java/large/src/cryptobench/verify/Determinism.java
package cryptobench.verify;

import cryptobench.core.HashFunction;

public class Determinism {
    public static boolean stable(HashFunction hash, String sample) {
        return hash.digest(sample) == hash.digest(sample);
    }

    public static boolean distinct(HashFunction hash, String left, String right) {
        if (left.equals(right)) {
            return true;
        }
        return hash.digest(left) != hash.digest(right);
    }
}
